src/lib/ProtocolAdapterFactory.ts

```diff
@@ -0,0 +1,29 @@
+import type {
+  Class,
+  Protocol,
+} from '@/types';
+import { BaseProtocolAdapter } from '@/protocols/BaseProtocolAdapter';
+import { AeternityAdapter } from '@/protocols/aeternity/libs/AeternityAdapter';
+import { BitcoinAdapter } from '@/protocols/bitcoin/libs/BitcoinAdapter';
+import {
+  PROTOCOL_AETERNITY,
+  PROTOCOL_BITCOIN,
+} from '@/constants';
+
+/**
+ * Encapsulates helper methods related to multiple protocol instances
+ */
+export const ProtocolAdapterFactory = (() => {
+  const protocolAdapterRegistry: Record<Protocol, Class<BaseProtocolAdapter> > = {
+    [PROTOCOL_AETERNITY]: AeternityAdapter,
+    [PROTOCOL_BITCOIN]: BitcoinAdapter,
+  };
+
+  function getAdapter(protocol: Protocol): BaseProtocolAdapter {
+    return new protocolAdapterRegistry[protocol]();
```

Comment:
This should not return new instance each time you run `getAdapter`. The instances should be stored. So I think oversimplified this now.

Reply:
Fixed it with a cache as discussed.